plugins/module_utils/wdc_redfish_utils.py

```diff
@@ -180,33 +187,60 @@ def _get_bundle_version(self,
         To determine if the bundle is multi-tenant or not, it looks inside the .bin file within the tarfile,
         and checks the appropriate byte in the file.
 
+        If not tarfile, the bundle is checked for 2048th byte to determine whether it is Gen2 bundle.
+        Gen2 is always single tenant at this time.
+
         :param str bundle_uri:  HTTP URI of the firmware bundle.
-        :return: Firmware version number contained in the bundle, and whether or not the bundle is multi-tenant.
-        Either value will be None if unable to determine.
+        :return: Firmware version number contained in the bundle, whether or not the bundle is multi-tenant
+        and bundle generation. Either value will be None if unable to determine.
         :rtype: str or None, bool or None
         """
         bundle_temp_filename = fetch_file(module=self.module,
                                           url=bundle_uri)
-        if not tarfile.is_tarfile(bundle_temp_filename):
-            return None, None
-        tf = tarfile.open(bundle_temp_filename)
-        pattern_pkg = r"oobm-(.+)\.pkg"
-        pattern_bin = r"(.*\.bin)"
         bundle_version = None
         is_multi_tenant = None
-        for filename in tf.getnames():
-            match_pkg = re.match(pattern_pkg, filename)
-            if match_pkg is not None:
-                bundle_version = match_pkg.group(1)
-            match_bin = re.match(pattern_bin, filename)
-            if match_bin is not None:
-                bin_filename = match_bin.group(1)
-                bin_file = tf.extractfile(bin_filename)
-                bin_file.seek(11)
-                byte_11 = bin_file.read(1)
-                is_multi_tenant = byte_11 == b'\x80'
-
-        return bundle_version, is_multi_tenant
+        gen = None
+
+        # If not tarfile, then if the file has "MMG2" or "DPG2" at 2048th byte
+        # then the bundle is for MM or DP G2
+        if not tarfile.is_tarfile(bundle_temp_filename):
+            bundle_file = open(bundle_temp_filename, "rb")
+            bundle_file.seek(2048)
+            cookie1 = bundle_file.read(4)
```

Comment:
```suggestion
            with open(bundle_temp_filename, "rb") as bundle_file:
                bundle_file.seek(2048)
                cookie1 = bundle_file.read(4)
```
Also you should add a check whether the file is at least 2048 + 4 bytes long before trying to read at that position.

Reply:
Implemented the comments, please review